[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74506
* __->__ #74505

Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35031523](https://our.internmc.facebook.com/intern/diff/D35031523)

jerryzh168 committed:
[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class

Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
regression tests

python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class"


Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35031523).

jerryzh168 committed:
Update on "[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class"


Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35031523](https://our.internmc.facebook.com/intern/diff/D35031523)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35031523).

vkuzo commented on torch/nn/qat/modules/conv.py:
```diff
@@ -85,7 +84,57 @@ def to_float(self):
+                 in_channels: int,
+                 out_channels: int,
+                 kernel_size: Tuple[int, ...],
+                 stride: Tuple[int, ...]=1,
```
nit: lint everywhere

jerryzh168 committed:
Update on "[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class"


Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35031523](https://our.internmc.facebook.com/intern/diff/D35031523)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class"


Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35031523](https://our.internmc.facebook.com/intern/diff/D35031523)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class"


Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35031523](https://our.internmc.facebook.com/intern/diff/D35031523)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class"


Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35031523](https://our.internmc.facebook.com/intern/diff/D35031523)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35031523).

jerryzh168 committed:
Update on "[quant][core] Refactor qat conv implementation to use the same _ConvNd as base class"


Summary:
att, this is to make supporting conv1d easier in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_conv_bn_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35031523](https://our.internmc.facebook.com/intern/diff/D35031523)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35031523).
